AstarteDeviceSDKCSharp/Device/AstarteDevice.cs

```diff
@@ -251,6 +251,22 @@ public void AddInterface(string astarteInterfaceObject)
 
         }
 
+        /// <summary>
+        /// Remove interface from device 
+        /// </summary>
+        /// <param name="interfaceName">Interface name</param>
+        public void RemoveInterface(string interfaceName)
+        {
+            AstarteInterface? astarteInterface = GetInterface(interfaceName);
+
+            if (astarteInterface == null)
+            {
+                throw new AstarteInterfaceException("Interface " + interfaceName + " not found");
+            }
+
+            _astarteInterfaces.Remove(astarteInterface.GetInterfaceName());
```

Comment:
When an interfaces has been removed from the device,introspection should also be re-sent  to Astarte. It is the same behavior as when adding an Interface. https://github.com/astarte-platform/astarte-device-sdk-csharp/blob/a858745ea51cc74f112b0b1ca21c7b3652eb1df0/AstarteDeviceSDKCSharp/Device/AstarteDevice.cs#L255

```suggestion
            _astarteInterfaces.Remove(astarteInterface.GetInterfaceName());
            
             if (IsConnected())
            {
                _astarteTransport.SendIntrospection();
            }
```

Reply:
I made changes to methods to remove an added interface.